numba/core/event.py

```diff
@@ -423,3 +429,53 @@ def on_exit(*exc_details):
 
         start_event(kind, data=data)
         yield
+
+
+def _get_native_ident():
+    try:
+        return threading.get_native_ident()
+    except AttributeError:
+        # Fallback for python <3.8
+        return threading.get_ident()
+
+
+def _prepare_chrome_trace_data(listener: RecordingListener):
+    """Prepare events in `listener` for serializing as chrome trace data.
+    """
+    # The spec for the trace event format can be found at:
+    # https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/edit   # noqa
+    # This code only uses the JSON Array Format for simplicity.
+    pid = os.getpid()
+    tid = _get_native_ident()
+    evs = []
+    for ts, rec in listener.buffer:
+        data = rec.data
+        cat = str(rec.kind)
+        ph = 'B' if rec.is_start else 'E'
+        name = data['name']
+        args = data
+        ev = dict(
+            cat=cat, pid=pid, tid=tid, ts=ts, ph=ph, name=name, args=args,
+        )
+        evs.append(ev)
+    return evs
+
+
+def _setup_chrome_trace_exit_handler():
+    """Setup a RecordingListener and a exit handler to write the captured events
+    to file.
+    """
+    listener = RecordingListener()
+    register("nb:run_pass", listener)
+    filename = config.CHROME_TRACE
+
+    @atexit.register
+    def _write_chrome_trace():
+        # The following output file is not multi-process safe.
+        evs = _prepare_chrome_trace_data(listener)
+        with open(filename, "w") as out:
+            json.dump(evs, out)
+
+
+if config.CHROME_TRACE:
```

Comment:
Should this be at module level?

Reply:
I wanted the env-var to enable the atexit hook and one can call `_setup_chrome_trace_exit_handler()` programmatically if the env-var is not set.